CONTRIBUTING.md

```diff
@@ -0,0 +1,221 @@
+if [ -z $2 ] ; then
+  BUILD_DIR="build-${SRC_DIR}"
+else
+  BUILD_DIR=$2
+fi
+echo "Build dir is $BUILD_DIR"
+if [ -d $BUILD_DIR ] ; then
+ echo "Build dir exists. Cmake will re-use cached data." 
+else
+ mkdir $BUILD_DIR
+fi
+cd $BUILD_DIR
+cmake $CMAKE_FLAGS $ABS_SRC_DIR
+make -j 16
+
+```
+
+Don't expect it will work from the first run as a charm. MySQL has a lot of dependencies on different libraries. So, you
+may use "try and fail" approach installing the development packages, or use some hints
+from [ documentation ](https://docs.percona.com/percona-server/8.0/compile-percona-server.html) .
+
+When the source finally successfully compiles, you're ready to run tests. You have to run at least the start and
+shutdown test of mysqld.
+
+```
+cd build-percona-server/mysql-test
+./mtr main.1st
+``` 
+
+When the tests are OK, you're ready to set up your favorite IDE for the project development. The build script above will
+give you a clue what CMake parameters to add to the project in IDE.
+
+To be able to run the mysqld binary in your IDE, first run the following command in the "mysql-test" directory of the
+build tree:
+
+```
+./mtr --manual-debug main.1st
+```
+
+Then find the string stating with "args:". This is the command line arguments you should pass to the execution of "
+mysqld" in your IDE. This step allows to run/debug "mysqld" from the IDE.
+
+From this moment you're ready to code and debug!
+
+### Testing
+
+Prior to submitting your code, ensure it is thoroughly tested, both manually and with automated test suites. The
+maintenance of high-quality code is paramount to ensure the longevity and functionality of our project.
+
+Testing of MySQL is a whole big thing, so please go to
+the [ official documentation on MySQL testing ](https://dev.mysql.com/doc/dev/mysql-server/latest/PAGE_TESTING_TOOLS.html)
+first.
+
+If it is bug fix, make sure that corresponding tests from the test suite work fine. It's a good idea to patch existing
+tests to cover your specific case. For the introduction of new features, the development of a comprehensive test suite
+is strongly recommended.
+
+## Preparing PR
+
+After finishing the tasks outlined above, you must prepare your branch for a pull request.   
+Please do not forget to run ``` git clang-format ``` on your code.
+
+Then, squash all commits to one or a couple of logically consistent commits.
+For example, the fix of bug should have just one commit. If you have a new feature that requires some changes to the
+existing code, then the first commit should contain required changes, and the second is a new feature and tests. It is
+not a dogma, on the other hand, we value a clean and consistent Git history. Please help us maintain this standard.
+
+Now you can push your local branch to your GitHub repository and make the pull request to the Percona repository.
+
+Please, write meaningful title for the PR and put the number of Jira issue at the beginning of the first line in the
```

Comment:
Ah. Here, we say they need a Jira number. 
Can we give them an example?

Reply:
Example added. Btw, above is a branch name example that contains Jire issue number also.